src/screens/Bridge/Exchange-CrossChain/Exchange.tsx

```diff
@@ -289,6 +293,15 @@ function Exchange({ fetchExchangeTitle }: Props) {
 
     const bestOffer = sortedOffers?.find((offer) => !!offer.feeInfo && offer.provider !== EXCHANGE_PROVIDER.LIFI);
 
+    if (hideAllOffers && selectedProvider) {
+      const selectedOffer = sortedOffers?.find((offer) => !!offer.feeInfo && offer.provider === selectedProvider);
+      if (selectedOffer) {
+        return sortedOffers.sort(function (x, y) {
+          return x.provider === selectedProvider ? -1 : y.provider === selectedProvider ? 1 : 0;
+        });
```

Comment:
Just a note that you can use a shorthand return here `=>` to simplify the code as opposed to nested returns.

Reply:
Updated it!